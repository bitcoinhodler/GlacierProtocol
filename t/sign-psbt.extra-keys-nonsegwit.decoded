{
  "txid": "c1c0d946d60474ba440717954df6147aa61825e895c6b5ace409691623e05d27",
  "hash": "c1c0d946d60474ba440717954df6147aa61825e895c6b5ace409691623e05d27",
  "version": 2,
  "size": 371,
  "vsize": 371,
  "weight": 1484,
  "locktime": 0,
  "vin": [
    {
      "txid": "fa1558b23e996a8b6ddc1bd3d0023e382e599fae7bae49da3e52d5519b105ab2",
      "vout": 1,
      "scriptSig": {
        "asm": "0 3044022036332e09987e0452666b10854b6c6493cdb57b0f8a8ecc35a9071e857878c13302206e06adb61b3cbdc5b0be5c523f7c9fe8864a2a7b5ca7b11a1b64b3f8ba6c198d[ALL] 304402205999776b1c1c3d42e838872532a2ec3af7946bd5c616b94485aa40ebcb9d2ee2022055e10486bf6975f3e4bb233e34c866f7b16071e32cd081b1e972bd7c8198d3f1[ALL] 522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae",
        "hex": "00473044022036332e09987e0452666b10854b6c6493cdb57b0f8a8ecc35a9071e857878c13302206e06adb61b3cbdc5b0be5c523f7c9fe8864a2a7b5ca7b11a1b64b3f8ba6c198d0147304402205999776b1c1c3d42e838872532a2ec3af7946bd5c616b94485aa40ebcb9d2ee2022055e10486bf6975f3e4bb233e34c866f7b16071e32cd081b1e972bd7c8198d3f1014c8b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae"
      },
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.42496290,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_HASH160 8546e747c8094376b0a0af44bc5bbb8dd4f0218d OP_EQUAL",
        "hex": "a9148546e747c8094376b0a0af44bc5bbb8dd4f0218d87",
        "reqSigs": 1,
        "type": "scripthash",
        "addresses": [
          "2N5PvoFj6adLPcqHAD24MxEZxvdQMGBLXow"
        ]
      }
    }
  ]
}
