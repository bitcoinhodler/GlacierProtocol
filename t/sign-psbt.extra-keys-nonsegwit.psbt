cHNidP8BAFMCAAAAAZPB7OKev90s415vOhJVp4jY7Gn+6LG0nVUiZhAYhyAzAQAAAAD/////ASJxiAIAAAAAF6kUhUbnR8gJQ3awoK9EvFu7jdTwIY2HAAAAAAABAIoCAAAAAT4FcJQhTvbeTJZn6Z+xxi4/sIQN6LErzGNZGqiJctqJAQAAABcWABSDE1uBYw40E8feHLAHGedlcYzTPf////8CEH6IAgAAAAAXqRQ/FDMcOPCztN3lWhSEj1cymX/jEYegf4gCAAAAABepFL05lruabEAdp+ZdTbs6m5eM0EXshwAAAAABBItSIQPRTdz7aBf1V5aVu7PrPhhYh78pQrAx5vcWNDuP5+no4iECj81G+GFLLL8xgJaWgkKh4ivPttjyttyTnIwnw0eyk3shAxWstVASD0zctGDVxJCAq1CMpMzY3t7Kwi/urIzQF9TBIQIrBj7i8i+eGYLBQP53hnLGgxEaeW3Mzb1Hu2Xj1gipg1SuAAA=
