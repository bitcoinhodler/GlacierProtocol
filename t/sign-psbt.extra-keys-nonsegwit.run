#!/bin/bash
set -e

# Test validation of extra keys. Only need 2, providing 4 as I would in the test withdrawal process


$GLACIERSCRIPT --regtest=$1 sign-psbt << INPUT 2>&1
y
y
y
y
y
y
../../t/sign-psbt.extra-keys-nonsegwit.psbt
y
4
cQCrT9Ncs9729ao7jbmAWrD9z7tF64s2yKzmD6nkiLAi9sXVZWAn
cP65UeSDZPiTLB6CBwasWv9oJYEjRgQXhswfwcT9HscEKDcEbgy4
cNYaH3onqrdMffpznhMMmrHn34fuTU59w5j8LM3H42VPcUsLeXy5
cRoydfinDRzzRQJp5niqJWukSYTfPJQM6ytqGN6nzonaz1mafgwD
y
INPUT
