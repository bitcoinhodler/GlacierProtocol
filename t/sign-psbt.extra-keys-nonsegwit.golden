Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?Input a filename located in the current directory which contains the PSBT:
0.42500000 BTC in unspent inputs from cold storage address 2NAVkgszAvjZZwWpHpbCaxSaNqCCXPD8Bts
0.42496290 BTC going to destination address 2N5PvoFj6adLPcqHAD24MxEZxvdQMGBLXow
Fee amount: 0.00003710
Confirm? (y/n): 
How many private keys will you be signing this transaction with (at least 2 required)?
#: Key #1: Key #2: Key #3: Key #4: 
Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.42500000 BTC in unspent inputs from cold storage address 2NAVkgszAvjZZwWpHpbCaxSaNqCCXPD8Bts
0.42496290 BTC going to destination address 2N5PvoFj6adLPcqHAD24MxEZxvdQMGBLXow
Fee amount: 0.00003710

Signing with private keys: 
cQCrT9Ncs9729ao7jbmAWrD9z7tF64s2yKzmD6nkiLAi9sXVZWAn
cP65UeSDZPiTLB6CBwasWv9oJYEjRgQXhswfwcT9HscEKDcEbgy4
cNYaH3onqrdMffpznhMMmrHn34fuTU59w5j8LM3H42VPcUsLeXy5
cRoydfinDRzzRQJp5niqJWukSYTfPJQM6ytqGN6nzonaz1mafgwD


Confirm? (y/n): 
Calculating transaction...

Final fee rate: 10 satoshis per vbyte

Raw signed transaction (hex):
02000000014c95f31df67bd1563ceb600a7c1a5613d9783aa21473b413166275bfbb69e43f01000000fd1e010047304402207c93b46aca78ddd0a7801f51cb65464a03a953a40f7a98311d093da63a7d56b102202c5cdfb36c49ecc7ab302e705d3ce2a4cbfb60d30f15da4cb1c20447c723a43501473044022037ca50526b9d5fcbe6b58334e88b361b8e43d0ef27e28315099265d84e90a64d02207e32224a392befb813025a19ab95ad07b2f9766c3e872dd0aa16ae6e3e6549ab014c8b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354aeffffffff01227188020000000017a9148546e747c8094376b0a0af44bc5bbb8dd4f0218d8700000000

Transaction fingerprint (md5):
afd029da36f71c193493e8ef518da711
QR code for transaction written to transaction.png
